src/mam4xx/modal_aer_opt.hpp

```diff
@@ -0,0 +1,1665 @@
+constexpr int refindex_im = 10;  // imaginary refractive index
+
+struct AerosolOpticsDeviceData {
+  // devices views
+  // FIXME: add description of these tables.
+  View1D refitabsw[ntot_amode][nswbands];
+  View1D refrtabsw[ntot_amode][nswbands];
+  View3D abspsw[ntot_amode][nswbands];
+  View3D asmpsw[ntot_amode][nswbands];
+
+  View1D refrtablw[ntot_amode][nlwbands];
+  View1D refitablw[ntot_amode][nlwbands];
+  View3D absplw[ntot_amode][nlwbands];
+  View3D extpsw[ntot_amode][nswbands];
+
+  ComplexView1D crefwlw;
+  ComplexView1D crefwsw;
+  ComplexView2D specrefindex_sw[ntot_amode];
+  ComplexView2D specrefindex_lw[ntot_amode];
+};
+
+inline void set_aerosol_optics_data_for_modal_aero_sw_views(
+    AerosolOpticsDeviceData &aersol_optics_data) {
+
+  for (int d1 = 0; d1 < ntot_amode; ++d1) {
+    for (int d5 = 0; d5 < nswbands; ++d5) {
+      aersol_optics_data.abspsw[d1][d5] =
+          View3D("abspsw", coef_number, refindex_real, refindex_im);
+      aersol_optics_data.extpsw[d1][d5] =
+          View3D("extpsw", coef_number, refindex_real, refindex_im);
+      aersol_optics_data.asmpsw[d1][d5] =
+          View3D("asmpsw", coef_number, refindex_real, refindex_im);
+
+      aersol_optics_data.refrtabsw[d1][d5] = View1D("refrtabsw", refindex_real);
+
+      aersol_optics_data.refitabsw[d1][d5] = View1D("refitabsw", refindex_im);
+    } // d5
+  }   // d1
+
+} // configure_aerosol_optics_data
+
+inline void set_aerosol_optics_data_for_modal_aero_lw_views(
+    AerosolOpticsDeviceData &aersol_optics_data) {
+
+  for (int d1 = 0; d1 < ntot_amode; ++d1)
+    for (int d5 = 0; d5 < nlwbands; ++d5) {
+      aersol_optics_data.absplw[d1][d5] =
+          View3D("absplw3", coef_number, refindex_real, refindex_im);
+      aersol_optics_data.refrtablw[d1][d5] = View1D("refrtablw", refindex_real);
+      aersol_optics_data.refitablw[d1][d5] = View1D("refitablw", refindex_im);
+    } // d5
+
+} // set_aerosol_optics_data_for_modal_aero_lw_views
+
+inline void
+set_complex_views_modal_aero(AerosolOpticsDeviceData &aersol_optics_data) {
+  for (int i = 0; i < ntot_amode; ++i) {
+    aersol_optics_data.specrefindex_sw[i] =
+        ComplexView2D("specrefindex_sw", max_nspec, nswbands);
+  }
+
+  for (int i = 0; i < ntot_amode; ++i) {
+    aersol_optics_data.specrefindex_lw[i] =
+        ComplexView2D("specrefindex_lw", max_nspec, nlwbands);
+  }
+
+  aersol_optics_data.crefwlw = ComplexView1D("crefwlw", nlwbands);
+  aersol_optics_data.crefwsw = ComplexView1D("crefwsw", nswbands);
+} // set_complex_views_modal_aero
+
+struct DiagnosticsAerosolOpticsSW {
+  // FIXME; add description of these diagnostics variables
+  ColumnView extinct; //        ! aerosol extinction [1/m]
+  ColumnView absorb;  //         ! aerosol absorption [1/m]
+  View0D aodnir;
+  View0D aoduv;
+  View1D dustaodmode; // ntot_amode
+  View1D aodmode;     // ntot_amode
+  View1D burdenmode;  // ntot_amode
+  View0D aodabsbc;
```

Comment:
Just curious--what is the purpose of a `View0D` as opposed to a struct field for an on-device struct?

Reply:
I delete this code.